client/signup/steps/woocommerce-install/transfer.tsx

```diff
@@ -10,6 +22,76 @@ interface Props {
 export default function Transfer( { goToStep }: Props ): ReactElement | null {
 	const { __ } = useI18n();
 
+	const siteId = useSelector( getSelectedSiteId ) as number;
+
+	const [ progress, setProgress ] = useState( 0 );
+	const [ error, setError ] = useState( { transferFailed: false, transferStatus: null } );
+
+	const dispatch = useDispatch();
+
+	// Select state
+	const fetchingTransferStatus = !! useSelector( ( state: AppState ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+	const transferStatus = useSelector( ( state: AppState ) =>
+		getAutomatedTransferStatus( state, siteId )
+	);
+
+	const transferFailed = useSelector( ( state: AppState ) => hasUploadFailed( state, siteId ) );
+
+	// Initiate Atomic transfer
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatch( fetchAutomatedTransferStatus( siteId ) ); // todo: can possibly drop this, polling should be kicked off by initiate call
+		dispatch( initiateThemeTransfer( siteId, null, 'woocommerce' ) );
+	}, [ siteId, dispatch ] );
+
+	// Watch transfer status
+	useEffect( () => {
+		if ( fetchingTransferStatus ) {
+			return;
+		}
+
+		switch ( transferStatus ) {
```

Comment:
todo: Don't like this much. The process is slow so we need something more than a spinner but I'm not sure this approach makes sense either.

Reply:
Some prior art in here https://github.com/Automattic/wp-calypso/pull/56997/ uses some acceleration math to make things smoother.

Reply:
Also, much better looking progress bar in here:

https://github.com/Automattic/wp-calypso/pull/57051/files#diff-d9bd8319fe61d2742e0a99a51a83f4f831414bae75fa6108c9e031f5da32a901R83-R111

(This PR included the new spinner graphic we should be using if required too)